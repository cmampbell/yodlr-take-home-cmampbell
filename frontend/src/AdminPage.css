.AdminPage-UserCards {
    height: 70vh;
    width: 1260px;
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
    justify-content: flex-start;
    align-content: flex-start;
}

h1 {
    padding-bottom: 3%;
}