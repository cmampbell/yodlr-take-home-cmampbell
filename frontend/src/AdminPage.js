import React from "react";
import { useLoaderData } from "react-router-dom";
import UserCard from "./UserCard";
import './AdminPage.css'

// admin page needs to return list of users
const AdminPage = () => {
    const users = useLoaderData();
    return (
        <div className="AdminPage">
            <h1>
                Admin Page
            </h1>
            <div className="AdminPage-UserCards">
                {users.map(user => <UserCard
                    user={user}
                    key={user.id}
                    state={user.state}
                />)}
            </div>
        </div>
    );
}

export default AdminPage;