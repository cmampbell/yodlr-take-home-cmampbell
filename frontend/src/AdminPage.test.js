import React from "react";
import renderWithRouter from "./renderWithRouter";
import AdminPage from "./AdminPage";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLoaderData: () => (
        [{ "id": 1, "email": "kyle@getyodlr.com", "firstName": "Kyle", "lastName": "White", "state": "active" },
        { "id": 2, "email": "jane@getyodlr.com", "firstName": "Jane", "lastName": "Stone", "state": "active" }]
    )
}));

describe('AdminPage tests', () => {
    it('should render without crashing', () => {
        // define a full route as the first argument
        renderWithRouter(<AdminPage />)
    })

    it('should match the snapshot', () => {
        const { asFragment } = renderWithRouter(<AdminPage />)

        expect(asFragment()).toMatchSnapshot();
    })
})