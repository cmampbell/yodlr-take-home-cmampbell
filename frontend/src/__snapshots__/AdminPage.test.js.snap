// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminPage tests should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="AdminPage"
  >
    <h1>
      Admin Page
    </h1>
    <div
      class="AdminPage-UserCards"
    >
      <div
        class="card border-light"
      >
        <div
          class="card-body"
        >
          <div
            class="active card-header"
          >
            Kyle White
          </div>
          <p
            class="card-text"
          >
            Email: kyle@getyodlr.com
          </p>
          <p
            class="card-text"
          >
            State: Active
          </p>
        </div>
      </div>
      <div
        class="card border-light"
      >
        <div
          class="card-body"
        >
          <div
            class="active card-header"
          >
            Jane Stone
          </div>
          <p
            class="card-text"
          >
            Email: jane@getyodlr.com
          </p>
          <p
            class="card-text"
          >
            State: Active
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
